### PR TITLE
Improve README.md

### PR DESCRIPTION
**Pull request by snmgian**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,122 +1,128 @@
...
+``` ruby
+gifter.call(day_of_week: :friday, amount: 314)
 
-# RuleSets can have multiple outcomes when
-# conditions obtain in a certain Context.
+# => 7
+```
 
-ruleset.outcome "I matched!"
+If no outcomes match, the default one is returned.
```
Comment (migl) [inline README.md]: "If no outcome matches..."

File: README.md [hunk 2]
```diff
@@ -1,122 +1,128 @@
...
-```
-#### Applying Rules
+ - all *All of the nested conditions must suffice*
+ - any *At least one of its nested conditions must suffice*
+ - assert *A context value must be truish*
+ - between *Evaluates that a context value must belong to a specified range*
+ - cond *At least one slice of two nested conditions must suffice*
+ - day_of_week *Evaluates that a Date/DateTime/Time weekday is matched*
+ - eq *Tests a context value for equality*
+ - except *Evalutes that a any of the nested conditions don't suffice*
```
Comment (migl) [inline README.md]: "Evaluates that all of the nested condition don't suffice"

Remember that this condition also takes an attribute to evaluate from the context and check that is not equal to a passed value.

File: README.md [hunk 3]
```diff
@@ -1,122 +1,128 @@
...
+ - greater_than *Tests that context value is greater than something*
+ - greater_than_or_equal *Tests that context value is greater or equal than something*
```
Comment (migl) [inline README.md]: "Tests that context value is greater than or equal to something"

File: README.md [hunk 4]
```diff
@@ -1,122 +1,128 @@
...
+ - in *A context value must belong to a specified list of values*
+ - in_cyclic_order *TBD*
+ - include *A context value must include a specified one*
```
Comment (migl) [inline README.md]: "The context attribute must include a specified value"

File: README.md [hunk 5]
```diff
@@ -1,122 +1,128 @@
...
+ - less_than_or_equal *Tests that context value is less than or equal to something*
+ - less_than *Tests that context value is less than something*
 
-Rules have a `#call` method that returns true when all conditions are met. `#call` takes one argument, a `Context`, that has values for the various conditions to be evaluated against. If all the conditions pass, `#call` returns `true` and otherwise `false`. Example:
+Conditions can be nested, in such case, to the condition be sufficed, the nested conditions must
```
Comment (migl) [inline README.md]: Conditions can be nested. In such case, for the nesting condition to be met, the nested conditions must also be met.

File: README.md [hunk 6]
```diff
@@ -1,122 +1,128 @@
...
+also suffice.
 
-```ruby
-rule = Ruy::Rule.new
-rule.eq 21, :age
-rule.call(Context.new({age: 21, name: 'Leah'})) # => true
+``` ruby
+between :amount, 0, 1000 do
```
Comment (migl) [inline README.md]:
```ruby
between 0, 100, :amount do
  eq :friday, :day_of_week
end
```

File: README.md [hunk 7]
```diff
@@ -1,122 +1,128 @@
...
-
-An `Outcome` is a type of rule (`class Outcome < Rule`) that emits a value when the rule is evaluated (via `#call`), instead of just returning `true`. Example:
-
-```ruby
-outcome = Ruy::Outcome.new({sample: :outcome_value})
-outcome.eq 21, :age
-outcome.(Ruy::VariableContext.new({age: 21}, {})) # => {sample: :outcome_value}
+``` ruby
+all do
+  between :amount, 0, 1000
```
Comment (migl) [inline README.md]: Same here.

File: README.md [hunk 8]
```diff
@@ -1,122 +1,128 @@
...
+```
 
-ruleset.(Ruy::VariableContext.new({age: 21}, {})) # => "Nothing matched"
+``` ruby
+gifter.call(day_of_week: :friday, amounts_average: -> { Stats::Amounts.compute_average })
 ```
+### Outcomes
+
+An outcome is the result of a successful ruleset evaluation. An outcome can also have nested
+conditions, in such case, if the conditions met the outcome's value is returned.
```
Comment (migl) [inline README.md]: "...if the conditions meet, the outcome value is returned."

File: README.md [hunk 9]
```diff
@@ -1,122 +1,128 @@
...
+
+A rulset can have multiple outcomes, the first matching one is returned.
```
Comment (migl) [inline README.md]: "A RuleSet can have multiple outcomes; the first one that matches is returned."

File: README.md (hunk 1 above)
Comment (snmgian) [inline README.md]: Done

File: README.md (hunk 2 above)
Comment (snmgian) [inline README.md]: Done, nested conditions evaluation is documented below.

File: README.md (hunk 3 above)
Comment (snmgian) [inline README.md]: Done

File: README.md (hunk 4 above)
Comment (snmgian) [inline README.md]: Done

File: README.md (hunk 5 above)
Comment (snmgian) [inline README.md]: Done

File: README.md (hunk 6 above)
Comment (snmgian) [inline README.md]: Done

File: README.md (hunk 7 above)
Comment (snmgian) [inline README.md]: Done

**Pull request merged**